src/web/src/app/space/add/space-key-validator.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { AbstractControl, AsyncValidator, FormControl } from '@angular/forms';
+import { map } from 'rxjs/operators';
+import { SpaceService } from '../service';
+
+@Injectable({
+	providedIn: 'root',
+})
+export class SpaceKeyValidation implements AsyncValidator {
+
+	constructor(private space: SpaceService) {}
+
+	validate = (control: AbstractControl) => {
+
+		const { value } = control;
+
+		return this.space.isExists(value)
+			.pipe(
+				map(res => {
+					if (res) {
+						return { 'keyExists': true};
+					}
+				}),
+			);
```

Comment:
Тут почти ничего не происходит, а столько отступов и новых строк. Давай пожалуйста сократим чуток. 
```suggestion
		return this.space.isExists(value)
			.pipe(map(res => res ? {keyExists: true} : null));
```

Reply:
Сократил, но не до конца, как в твоём предложенной коде. Без ретурна на нижнем уровне не захотело работать так же, как и с тернарным оператором.